### ISSUE TITLE
UNUSED_SIGNAL warning if signal is only used via emit_signal() 

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot#89632]
### Tested versions

v4.3.dev5 
not reproducable in v4.3.dev4 

### System information

macOS 14.4

### Issue description

If you emit signals using emit_signal("signal_name") this will no longer register in 4.3dev5 as using this signal and trigger an UNUSED_SIGNAL warning for the signal. 

### Steps to reproduce

Write a simple script like

```
extends Node

signal value_changed()

var value:bool = false:
	set(new_value):
		if (new_value != value):
			value = new_value
			emit_signal("value_changed")
```

And observer the warning:
`Line 3 (UNUSED_SIGNAL):The signal "value_changed" is declared but never explicitly used in the class.`

If you instead use `value_changed.emit()` the warning is not shown.

If the script is used in the project you'll get a similar warning in the debug output running the game. 

### Minimal reproduction project (MRP)

See script lines above.

**Comment by dalexeev** [comment]
Yes, this is expected, which is why we added "explicitly" to the warning message. Note that you can use `Signal.emit()` (instead of `Object.emit_signal()`), which is a more recommended option.

We could add an exception for `emit_signal(constant_string)`, however this will not work in case of dynamic expressions and something like `call("emit_signal", "my_signal")`.

> This bug is also seems new to 4.3dev5 (or a regression), it's not present in 4.3dev4 or previous 4.x versions

It's a bug fix, the `UNUSED_SIGNAL` warning was never  produced before (in 4.x).

**Comment by lostminds** [comment]
I see, but then I think it could be a good idea to either add an exception for `emit_signal("signal_name")` for the specific signal (which I think could be seen as explicitly using the signal, even if technically it might not be) or be more clear in the warning that by explicit you mean using `signal_name.emit()` and that this is recommended. 

Otherwise I think you might get a lot more confused bug reports like this one once 4.3 is released and more people start getting these warnings that were not they may perceptive as incorrect just like I did.

**Comment by kuligs2** [comment]
Uhm, yeah its still annoying even when for example you try to emit signals inside a new thread. You cant use `some_signal.emit()`, you need to `call_deferred("emit_signal","some_signal",some_var)`


>Godot v4.3.beta3 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA RTX A2000 (NVIDIA; 31.0.15.5186) - Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz (8 Threads)

For example:
Note: Replace `2 == 3` with any of your own condition to emit the signal
```
extends Node

signal serverlist_updated(list_var)
var g_thread :Thread
var threading:bool = false

var list_var = {}
func g_loop():
	
	while threading:
		if 2 == 3:
			call_deferred("emit_signal","serverlist_updated",list_var)
		pass	
			
# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	g_thread = Thread.new()
	g_thread.start(g_loop,Thread.PRIORITY_HIGH)
				
	pass # Replace with function body.

```

**Comment by BrastenXBL** [comment]
This warning triggers when the Signal is connected from the Editor Signals panel. Perhaps the Project Settings default should be set to Ignore instead of Warn. Most new Godot users will be using the Editor panel, and will be confused by the warning. I'm already starting to see this show up in the community Reddit, and expect more instances once 4.3.0 is fully released.

**Comment by kuligs2** [comment]
> This warning triggers when the Signal is connected from the Editor Signals panel. Perhaps the Project Settings default should be set to Ignore instead of Warn. Most new Godot users will be using the Editor panel, and will be confused by the warning. I'm already starting to see this show up in the community Reddit, and expect more instances once 4.3.0 is fully released.

i prefer not to use editor to connect stuff that way because later on you can forget what was connected to where, as the codebase gets more convoluted and overall it would result in a mess.. i did that with input_action bindings... never again.. spent too many hours trying to debug why it dont work when it turned out i have previously bound the button so some different action.. it was a pain to debug.. Thats why i always try to do everything in code if possible.. that way all you have to do is read through it and youll see the whole picture on how everything is connected and works..

**Comment by TranquilMarmot** [comment]
Commented on the other issue, but I'll paste it here as well.

You can explicitly ignore the warning for a specific signal if you want:

```gdscript
@warning_ignore("unused_signal")
signal some_signal
```

https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/warning_system.html

I do this in spots where I _know_ that the signal is called from elsewhere.

If you always call signals like...
```gdscript
some_node.some_signal.emit()
```
You can find all references to the signal easily in the IDE.

**Comment by vittorio** [comment]
> You can explicitly ignore the warning for a specific signal if you want:
> 
> @warning_ignore("unused_signal")

It might be a good idea to add  directives like @warning_ignore_start("unused_signal") @warning_ignore_end(("unused_signal") or @warning_ignore_file("unused_signal")?

I don't want to ignore this warning completely, but for specific cases (like EventBus eg) i don't need it
So i have to place ignorance before each signal which doubles amount of code lines

as is
```GDScript
@warning_ignore("unused_signal")
signal ui_money_show
@warning_ignore("unused_signal")
signal ui_money_hide
@warning_ignore("unused_signal")
signal ui_money_update(value: int)

@warning_ignore("unused_signal")
signal ui_quest_show
@warning_ignore("unused_signal")
signal ui_quest_hide
@warning_ignore("unused_signal")
signal ui_quest_update(value: String)
```

to be
```GDScript
@warning_ignore_file("unused_signal")
signal ui_money_show
signal ui_money_hide
signal ui_money_update(value: int)

signal ui_quest_show
signal ui_quest_hide
signal ui_quest_update(value: String)
```

**Comment by kleonc** [comment]
> It might be a good idea to add directives like @warning_ignore_start("unused_signal") @warning_ignore_end(("unused_signal") or @warning_ignore_file("unused_signal")?

@vittorio There are [`@warning_ignore_start`](https://docs.godotengine.org/en/4.5/classes/class_@gdscript.html#class-gdscript-annotation-warning-ignore-start) and [`@warning_ignore_restore`](https://docs.godotengine.org/en/4.5/classes/class_@gdscript.html#class-gdscript-annotation-warning-ignore-restore) available since 4.4.dev7 (added in #76020).

**Comment by vittorio** [comment]
great, thank you
